### ISSUE TITLE
Apache 2.4.49 not working correctly with php-fpm socket

### ISSUE THREAD

**Issue by JohnRDOrazio** [oerdnj/deb.sury.org#1643]
**Frequently asked questions**

- [x] I have read [Frequently Asked Questions](https://github.com/oerdnj/deb.sury.org/wiki/Frequently-Asked-Questions)

**Describe the bug**

Please understand that this issue tracker is specifically meant to report issues with the packages and with the repositories. I won't teach you the basics of system administration, I won't help you with bugs in the software itself. For both, there are better sites where are people with more free time that allows them to be more friendly to newcomers.

Please make sure you read [Dear anonymous internet user asking for help...](https://berthub.eu/articles/posts/anonymous-help/) by Bert Hubert and [How to Report Bugs Effectively](https://www.chiark.greenend.org.uk/~sgtatham/bugs.html) by Simon Tatham.

If you understand the purpose of this issue tracker and describe your **problem** accurately (where the template below will help), I might be able to help you.

**To Reproduce**
1. On Ubuntu 20.04, issuing `sudo apt update` and `sudo apt upgrade`, **apache** get's updated to **v2.4.49**
2. **nginx** service is correctly running, **apache2** service is correctly running, **php8.0-fpm** service is correctly running
3. For good measure, restart all these services, they all restart correctly
4. All websites are giving 500 Internal server errors. The general system logs in my case have no useful information (`/var/log/syslog`, `/var/log/nginx/error.log`, `/var/log/apache2/error.log`), however the specific logs for each of my websites are showing a number of errors like this:
    ```
    AH10292: Invalid proxy UDS filename (proxy:unix:///var/www/vhosts/system/mydomain.com/php-fpm.sock|fcgi://127.0.0.1:9000/var/www/vhosts/mydomain.com/httpdocs/wp-login.php)
    ```
5. The `php-fpm.sock` file does exist in that path, and has correct permissions, and was working fine until before the package upgrade

**Expected behavior**
Apache should work fine with the php-fpm socket and websites should be running.

**Distribution (please complete the following information):**
 - OS: **Ubuntu 20.04.3 LTS**
 - Architecture: **amd64**
 - Repository: **ppa:ondrej/apache2**

**Package(s) (please complete the following information):**
Sorry I already removed **apache 2.4.49**, removed the **ppa:ondrej/apache2** repository, and reinstalled apache which is now at **v2.4.41**, and all my websites are back up and running again.

**Additional context**
I use Plesk web admin edition to help manage my server and my websites: **Plesk Obsidian 18.0.38 Update #2**, last updated on Sept 15, 2021 02:19 AM.

Other pleskians had the same problem, as can be seen on this thread: https://talk.plesk.com/threads/all-domains-apache2-error-500.362208/ .

**Comment by oerdnj** [comment]
There were not any updates to the packaging, so this needs to be reported to Apache upstream.

**Comment by JohnRDOrazio** [comment]
Ok I wrote to the Apache mailing list, and got this answer:

> I guess that it is fixed by r1893519 (trunk) [[1](https://svn.apache.org/viewvc?view=revision&revision=1893519)]
> This has been backported in r1893559 (2.4.x) [[2](https://svn.apache.org/viewvc?view=revision&revision=1893559)] and will be part of 2.4.50.

Seems to make sense, r1893519 is:

> mod_proxy: Handle UDS URIs with empty hostname as if they had no hostname.
> 
> It was reported to me (privately) that r1893101 broke existing settings like:
>   SetHandler "proxy:unix:///path/to/uds.sock|fcgi://localhost/"
> 
> RFC 3986 (section 3.2.2) says that:
>    If the URI scheme defines a default for host, then that default
>    applies when the host subcomponent is undefined or when the
>    registered name is empty (zero length).  For example, the "file" URI
>    scheme is defined so that no authority, an empty host, and
>    "localhost" all mean the end-user's machine, whereas the "http"
>    scheme considers a missing authority or empty host invalid.
> 
> Let's consider that the "unix" scheme is closer to the "file" scheme than
> the "http" one, and accept "unix:///path/to/uds.sock" as a valid URI.

So I guess we just have to wait for 2.4.50 to come out.

**Comment by oerdnj** [comment]
That's definitely the same issue, and also irrelevant here as "Server version: Apache/2.4.29 (Ubuntu)" is not provided by this repository.

**Comment by ghubz** [comment]
After upgrading to **2.4.49-3+ubuntu18.04.1+deb.sury.org+2** the same issue still occurs.

**Comment by oerdnj** [comment]
Then there must be multiple issues and they need to be reported separately to the Apache upstream.

**Comment by JohnRDOrazio** [comment]
> After upgrading to **2.4.49-3+ubuntu18.04.1+deb.sury.org+2** the same issue still occurs. Downgrading to 2.4.48 is not an option so let's hope 2.4.50 comes out soon. For the time being apache is set to automatically restart if unresponsive for 1 minute.

In fact the issue was originally reported for v2.4.49 in this thread, and was for this version that I reported to the Apache mailing list, where the issue was confirmed for v2.4.49.
I've tried downgrading to the system version (currently v2.4.41), but it has only brought that many more problems... Downgrading is not fun. I really hope v2.4.50 comes out soon.
Apache has been stable for me for years now, I haven't had any problems for almost 10 years. This is pretty serious... I realize on one hand that you can't demand from an open source project that's free, but when a good portion of websites around the world stay depending on such a project, then these kinds of issues can have a very big impact. If they realize this over at Apache, they'll push v2.4.50 as soon as possible I hope

**Comment by mowny** [comment]
> After upgrading to **2.4.49-3+ubuntu18.04.1+deb.sury.org+2** the same issue still occurs. Downgrading to 2.4.48 is not an option so let's hope 2.4.50 comes out soon. For the time being apache is set to automatically restart if unresponsive for 1 minute.

That has to be a completely different problem as the problem discussed here won't go away on a restart.

@oerdnj did you add r1893559 to your build yesterday? I'm seeing success reports in the plesk forum
@JohnRDOrazio if he did, then you can go back to this repo

**Comment by ghubz** [comment]
@mowny My issue (which has also been reported on other forums) is that apache will randomly hang without any error, even in debug mode there is no obvious error of what happened. This started right after upgrading to 2.4.49. Running 2.4.49 with PHP-FPM 7.4. Restarting the service only mitigates the problem until the next time. It happens 4-5 times a day. The only thing the error log says is [mpm_event:notice] [pid 6419:tid 139873254071790] AH00491: caught SIGTERM, shutting down

Yes most probably is a different issue than yours but it's still an issue related to 2.4.49. 

@JohnRDOrazio Open source projects of this magnitude (half of the internet relying on them) come with responsibility so yes, I would expect it to be addressed fairly quickly.

**Comment by JohnRDOrazio** [comment]
> @oerdnj did you add r1893559 to your build yesterday? I'm seeing success reports in the plesk forum @JohnRDOrazio if he did, then you can go back to this repo

I just readded the repo, and issued the update, and everything seems to be working again. So I'm guessing @oerdnj did take it on himself to pull the fix. Thank you so much! That saves so much pain.

**Comment by oerdnj** [comment]
> > After upgrading to **2.4.49-3+ubuntu18.04.1+deb.sury.org+2** the same issue still occurs. Downgrading to 2.4.48 is not an option so let's hope 2.4.50 comes out soon. For the time being apache is set to automatically restart if unresponsive for 1 minute.
> 
> 
> 
> That has to be a completely different problem as the problem discussed here won't go away on a restart.
> 
> 
> 
> @oerdnj did you add r1893559 to your build yesterday? I'm seeing success reports in the plesk forum

Yes, and two more related patches.

**Comment by oerdnj** [comment]
This bug has been fixed and the other one is tracked as #1646, so closing.
